### ISSUE TITLE
unable to install angular-cli 

### ISSUE THREAD

**Issue by findeiskahori** [npm/npm#14196]
I am unable to install angular-cli.

And when I enter in prompt npm instal -g angular-cli

<details>

```
C:\Windows\system32>npm install -g angular-cli
npm ERR! git clone --template=C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: Cloni
ng into bare repository 'C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660ac
c6a2a6453f54-e286d0b5'...
npm ERR! git clone --template=C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: ssh:
connect to host github.com port 22: Connection timed out
npm ERR! git clone --template=C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: fatal
: Could not read from remote repository.
npm ERR! git clone --template=C:\Users\findeis.kahori\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5:
npm ERR! git clone --template=C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: Pleas
e make sure you have the correct access rights
npm ERR! git clone --template=C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: and t
he repository exists.
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "angular-cli"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true clone --template=C:\Users\findeis.kahori\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:web
pack/compression-webpack-plugin.git C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b
96d25a660acc6a2a6453f54-e286d0b5
npm ERR! Cloning into bare repository 'C:\Users\myusername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e
91b96d25a660acc6a2a6453f54-e286d0b5'...
npm ERR! ssh: connect to host github.com port 22: Connection timed out
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR!
```

</details>
### supporting information:
- `npm -v` prints:  3.10.3
- `node -v` prints: v6.7.0
- `npm config get registry` prints: https://registry.npmjs.org/
- Windows, OS X, or Linux?: Windows 7
- Network issues:
  - Geographic location where npm was run:
  - [X] I use a proxy to connect to the npm registry.
  - [ ] I use a proxy to connect to the web.
  - [X] I use a proxy when downloading Git repos.
- `npm config list` prints:

<details>

```
; cli configs
user-agent = "npm/3.10.3 node/v6.7.0 win32 x64"

; userconfig C:\Users\findeis.kahori\.npmrc
https-proxy = "http://myusername:password@proxy.server.com:8080/"
proxy = "http://myusername:password@proxy.server.com:8080/"
strict-ssl = false

; builtin config undefined
https-proxy http://proxy.swm.de:8080 = true
prefix = "C:\\Users\\myusername\\AppData\\Roaming\\npm"
proxy http://proxy.swm.de:8080 = true
strict-ssl = false

; node bin location = C:\Program Files\nodejs\node.exe
; cwd = C:\Windows\system32
; HOME = C:\Users\myusername
; "npm config ls -l" to show all defaults.

 `git config --global -l `  prints:
user.name=[gitUsername]
user.email=[my e-mail address]
http.proxy=http://[proxyUsername]:[ProxyPassword]@proxy.server.com:8080
http.sslverify=false
https.proxy=https://[proxyUsername]:[ProxyPassword]@proxy.server.com:8080
https.sslverify=false
url.https://github.com.insteadof=git://github.com
url.http://github.com.insteadof=git://github.com
```

</details>

[npm-debug_copy.txt](https://github.com/npm/npm/files/513408/npm-debug_copy.txt)

---

Thank you in advance for your help!

Kind regards,

Kahori

**Comment by khanujasunny** [comment]
Same for me... was you able to solve this issue ?

**Comment by timadamson** [comment]
Also having this problem

**Comment by findeiskahori** [comment]
@khanujasunny 
Not yet.  I am still waiting for solutions.

**Comment by chrislow** [comment]
Same error here.

* npm: 3.10.8
* node: v6.6.0

**Comment by pphil3** [comment]
I also have the same issue
node v6.7.0
npm  v3.10.3

**Comment by chrislow** [comment]
Looks like installing the latest git version fixes this error. 
Go and download it from here: https://git-scm.com/

Current running version: `git --version`: git version 2.10.1.windows.1

**Comment by findeiskahori** [comment]
Hi Chrislow,

I installed 2.10.1 and tried again, but it does not work.

git --version prints now       git version 2.10.1.windows.1



C:\Windows\system32>npm install -g angular-cli
npm WARN addRemoteGit Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Socket.<anonymous> (internal/child_process.js:334:11)
npm WARN addRemoteGit     at emitOne (events.js:96:13)
npm WARN addRemoteGit     at Socket.emit (events.js:188:7)
npm WARN addRemoteGit     at Pipe._handle.close [as _onclose] (net.js:493:12)
npm WARN addRemoteGit  webpack/compression-webpack-plugin#7e55907cd54a2e91b96d25a660acc6a2a6453f54 resetting remote C:\Users\MyUsername\AppData\Roaming\npm-cache\_git
-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-ec76b2fe because of error: { Error: Command failed: git -c core.lo
ngpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Socket.<anonymous> (internal/child_process.js:334:11)
npm WARN addRemoteGit     at emitOne (events.js:96:13)
npm WARN addRemoteGit     at Socket.emit (events.js:188:7)
npm WARN addRemoteGit     at Pipe._handle.close [as _onclose] (net.js:493:12)
npm WARN addRemoteGit   killed: false,
npm WARN addRemoteGit   code: 1,
npm WARN addRemoteGit   signal: null,
npm WARN addRemoteGit   cmd: 'git -c core.longpaths=true config --get remote.origin.url' }
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.com/webpack/compression-webpack-plugin.git C
:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-ec76b2fe: Clo
ning into bare repository 'C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660
acc6a2a6453f54-ec76b2fe'...
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.com/webpack/compression-webpack-plugin.git C
:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-ec76b2fe: fat
al: unable to connect to github.com:
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.com/webpack/compression-webpack-plugin.git C
:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-ec76b2fe: git
hub.com[0: 192.30.253.113]: errno=No error
npm WARN addRemoteGit Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit  webpack/compression-webpack-plugin#7e55907cd54a2e91b96d25a660acc6a2a6453f54 resetting remote C:\Users\MyUsername\AppData\Roaming\npm-cache\_git
-remotes\git-https-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-541c96f3 because of error: { Error: Command failed: git -c c
ore.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit   killed: false,
npm WARN addRemoteGit   code: 1,
npm WARN addRemoteGit   signal: null,
npm WARN addRemoteGit   cmd: 'git -c core.longpaths=true config --get remote.origin.url' }
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror https://github.com/webpack/compression-webpack-plugin.git
 C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-541c9
6f3: Cloning into bare repository 'C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a
2e91b96d25a660acc6a2a6453f54-541c96f3'...
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror https://github.com/webpack/compression-webpack-plugin.git
 C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-541c9
6f3: fatal: unable to access 'https://github.com/webpack/compression-webpack-plugin.git/': Failed to connect to github.com port 443: Timed out
npm WARN addRemoteGit Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit  webpack/compression-webpack-plugin#7e55907cd54a2e91b96d25a660acc6a2a6453f54 resetting remote C:\Users\MyUsername\AppData\Roaming\npm-cache\_git
-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5 because of error: { Error: Command failed: git -c core.lo
ngpaths=true config --get remote.origin.url
npm WARN addRemoteGit
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit   killed: false,
npm WARN addRemoteGit   code: 1,
npm WARN addRemoteGit   signal: null,
npm WARN addRemoteGit   cmd: 'git -c core.longpaths=true config --get remote.origin.url' }
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: Cloni
ng into bare repository 'C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660ac
c6a2a6453f54-e286d0b5'...
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: ssh:
connect to host github.com port 22: Connection timed out
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: fatal
: Could not read from remote repository.
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5:
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: Pleas
e make sure you have the correct access rights
npm ERR! git clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:webpack/compression-webpack-plugin.git C:\
Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b96d25a660acc6a2a6453f54-e286d0b5: and t
he repository exists.
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "angular-cli"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true clone --template=C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git@github.com:web
pack/compression-webpack-plugin.git C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e91b
96d25a660acc6a2a6453f54-e286d0b5
npm ERR! Cloning into bare repository 'C:\Users\MyUsername\AppData\Roaming\npm-cache\_git-remotes\git-github-com-webpack-compression-webpack-plugin-git-7e55907cd54a2e
91b96d25a660acc6a2a6453f54-e286d0b5'...
npm ERR! ssh: connect to host github.com port 22: Connection timed out
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR!
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Windows\system32\npm-debug.log

**Comment by gual** [comment]
I was having this same error using Babun, I tried with Windows PowerShell and it worked. It could be a workaround.

**Comment by cognitivedesire** [comment]
Did anyone solve this

**Comment by chrislow** [comment]
I can only write down the steps i did to make it work:
- [x] Uninstall Git, Node.js, npm
- [x] Install Git 
- [x] Install Node.js witn npm
- [x] `npm install anuglar-cli -g `

Try to install the latest versions.
I got: 
* node --version v5.12.0
* npm -v 3.10.5
* git --version git version 2.8.1.windows.1

**Comment by codernonkey** [comment]
I had the same problem: npm install with a proxy requiring authentication. Some hints which helped me:

- I can recommend running npm with --verbose (**npm --verbose -g install angular-cli**): I had the case that without verbose git was reporting the last error which was a ssh/port 22 connection issue. However, before that it hat tried to connect with https and was getting different errors which were guiding me in the right directions.

It seems when calling git from npm some (?) of the git config --global settings are deleted/ignored, especially http.sslVerfiy and https.proxy. 

- For the first (http.sslVerify) I had to add the CA of the proxy to the git config. For this:

1. Get the CA of your proxy. One simple approach: open https://github.com in firefox, click on the cert info and let disply the certificate (some hopefully obvious clicks), in the details section you see the chain; click on the CA (root) of the chain and export it as X.509 PEM cert (.crt)
2. Find out where your git stores its CA infos by git config --list | grep http.sslcainfo, e.g., C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
3. Edit the ca-bundle.crt file (you will require admin privileges) and add the content of your exported CA.crt file (both sould be in the format -----BEGIN CERTIFICATE----- <base64>....)


- For the second (https.proxy) I found a hint on https://github.com/npm/npm/issues/3559: you have to set the env var to make the settings available to git. On Windows:
set HTTP_PROXY=http://<proxy user name>:<URL encoded password>@<proxy server>:<proxy port>
set HTTPS_PROXY=%HTTP_PROXY%
setx HTTP_PROXY %HTTP_PROXY%
setx HTTPS_PROXY %HTTP_PROXY%

 
To summarize, I had executed:

npm config set proxy %HTTP_PROXY%
npm config set https-proxy %HTTP_PROXY%
npm config set strict-ssl false
npm set registry https://registry.npmjs.org/

git config --global https.proxy %HTTP_PROXY%
git config --global http.proxy %HTTP_PROXY%
git config --global url."https://".insteadOf git://
git config --global url."https://github.com/".insteadOf git@github.com:

**Comment by devendrapratapsingh** [comment]
To fix this issue, I followed below steps:

Check if you working behind a proxy server, If yes then

In windows command prompt, set your proxy
set HTTP_PROXY=http://[proxy_user_name]:[password]@[proxy_server]:[proxy_port]
set HTTPS_PROXY=%HTTP_PROXY%
but above setting will for that particular window

after that set the proxy for npm like below
npm config set proxy http://[proxy_user_name]:[password]@[proxy_server]:[proxy_port]
npm config set https_proxy http://[proxy_user_name]:[password]@[proxy_server]:[proxy_port]

once you set above proxy setting for windows command prompt and npm, you are all set to install angular-cli

run below command to install:

npm install -g angular-cli
let me know, if this doesn't work for windows, for the mac also use equivalent command to this.

**Comment by pphil3** [comment]
Thank you @codernonkey! To follow your steps finally solve all the issues for me (I am behind an entreprise proxy)

At first, I skipped to add the CA of the proxy to the git config, but once I also did that, now the install is working.

Note: To export the Cert from  https://github.com, I used Chrome instead of Firefox; Chrome exports in a binary format, so you need had to convert the exported cert from binary to text (openssl, or some online conveter tool). If you want to avoid this extra step, I guess Firefox is the way.

Clever explanations, thank you again!
